documentation/collectionruleexamples.md

```diff
@@ -0,0 +1,261 @@
+# Collection Rule Examples
+
+The following examples provide sample scenarios for using a collection rule. These templates can be copied directly into your configuration file with minimal adjustments to work with your application (for more information on configuring an egress provider, see [egress providers](./configuration.md#egress-configuration)), or they can be adjusted for your specific use-case. [Learn more about configuring collection rules](collectionrules.md).
+
+## Collect Trace - Startup (`Startup` Trigger)
+
+### JSON
+
+```json
+{
+  "AssemblyLoadTraceOnStartup": {
+    "Trigger": {
+      "Type": "Startup"
+    },
+    "Actions": [
+      {
+        "Type": "CollectTrace",
+        "Settings": {
+          "Providers": [{
+              "Name": "Microsoft-Windows-DotNETRuntime",
+              "EventLevel": "Informational",
+              "Keywords": "0x8"
+          }],
+          "Duration": "00:00:15",
+          "Egress": "artifacts"
+        }
+      }
+    ]
+  }
+}
+```
+
+### Explanation
+
+This rule, named "AssemblyLoadTraceOnStartup", will trigger on a process's startup. When the rule is triggered, a trace will be collected for 15 seconds and egressed to the specified `Egress` provider (in this case, `artifacts` has been configured to save the trace to the local filesystem). The trace will capture events from an event provider named `Microsoft-Windows-DotNETRuntime`, and will collect events at or above the `Informational` level using the keyword `0x8` (`LoaderKeyword`). The trace will request rundown by default, and the `BufferSizeInMB` has the default value of 256 MB.
+
+## Collect GCDump - Heap Size (`EventCounter` Trigger)
+
+### JSON
+
+```json
+{
+  "LargeGCHeapSize": {
+    "Trigger": {
+      "Type": "EventCounter",
+      "Settings": {
+        "ProviderName": "System.Runtime",
+        "CounterName": "gc-heap-size",
+        "GreaterThan": 1
```

Comment:
nit: won't 1 trigger right away? Maybe something a little larger?

Reply:
I bumped it up to 10 MB after a little playing around, though I still don't have much intuition for this. Let me know if you want me to change this before I check it in (and if not, we can always update the example later if people prefer another value).